pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -598,43 +598,60 @@ func verifyStatus(ctx *domainContext, status *types.DomainStatus) {
 		status.DomainId = 0
 		publishDomainStatus(ctx, status)
 	} else {
-		if !status.Activated {
-			log.Warnf("verifyDomain(%s) domain came back alive; id  %d\n",
-				status.Key(), domainID)
-			status.LastErr = ""
-			status.LastErrTime = time.Time{}
-			status.DomainId = domainID
-			status.BootTime = time.Now()
-			log.Infof("Update domainId %d bootTime %s for %s",
-				status.DomainId, status.BootTime.Format(time.RFC3339Nano),
-				status.Key())
-			status.Activated = true
-			status.State = types.RUNNING
-			publishDomainStatus(ctx, status)
-		} else if domainID != status.DomainId {
-			// XXX shutdown + create?
-			log.Warnf("verifyDomain(%s) domainID changed from %d to %d\n",
-				status.Key(), status.DomainId, domainID)
-			status.DomainId = domainID
-			status.BootTime = time.Now()
-			log.Infof("Update domainId %d bootTime %s for %s",
-				status.DomainId, status.BootTime.Format(time.RFC3339Nano),
-				status.Key())
-			publishDomainStatus(ctx, status)
-		}
-		// check if qemu processes has crashed
-		hasQemu := status.VirtualizationMode == types.HVM || status.VirtualizationMode == types.FML || status.IsContainer
-		if configActivate && status.Activated && hasQemu && !hyper.IsDeviceModelAlive(status.DomainId) {
-			errStr := fmt.Sprintf("verifyStatus(%s) qemu crashed",
+		if !hyper.IsDomainHealthy(status.DomainId) {
+			errStr := fmt.Sprintf("verifyStatus(%s) domain is unhealthy",
```

Comment:
We don't need this change. 

I just wanted one function call and check just before the hasQemu code.

Reply:
Removed. 
Calling the same in the if statement before we do qemu check